controllers/managedocs_controller.go

```diff
@@ -1598,6 +1599,27 @@ func (r *ManagedOCSReconciler) reconcileEgressNetworkPolicy() error {
 			)
 		}
 
+		if r.DeploymentType == consumerDeploymentType {
+
+			storageProviderEndpointURL := r.addonParams[storageProviderEndpointKey]
+			storageProviderEndpointHost := strings.Split(storageProviderEndpointURL, ":")[0]
+
+			ip := net.ParseIP(storageProviderEndpointHost)
+
+			if ip == nil {
```

Comment:
I think you wanted `!=` and not `==`.
Also, let's not pollute the func scope if we don't have to.
```suggestion
			if ip := net.ParseIP(storageProviderEndpointHost); ip == nil {
```

Reply:
I want to check if the provided storageProviderEndpointHost is a nodeIP or a hostName, we want to allow the loadBalancer Endpoint HostName, we can ignore if we have nodeIP as it is already reachable.